save-preprocessor/src/main/kotlin/com/saveourtool/save/preprocessor/service/TestsPreprocessorToBackendBridge.kt

```diff
@@ -58,45 +61,49 @@ class TestsPreprocessorToBackendBridge(
                 )
             )
         }
-        .bodyToMono()
+        .blockingBodyToMono()
 
     /**
      * @param sourceId
      * @param commitId
      * @return [TestsSourceSnapshotDto] found by provided values
      */
+    @NonBlocking
     fun findTestsSourceSnapshot(sourceId: Long, commitId: String): Mono<TestsSourceSnapshotDto> = webClientBackend.get()
         .uri("/test-suites-sources/find-snapshot?sourceId={sourceId}&commitId={commitId}", sourceId, commitId)
         .retrieve()
-        .bodyToMono()
+        .blockingBodyToMono()
```

Comment:
why blockingBodyToMono?

Reply:
A network operation (from the _preprocessor_ to the _backend_) is a blocking I/O operation, just like any other I/O operation (e.g.: a file system access).

This change offloads network requests the preprocessor **initiates** (not the ones it **receives**) from the `reactor-http-*` pool (which serves the **received** requests) to the `bounded-elastic-*` pool.

Reply:
it means that all network operation should be blockingBodyToMono?